fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -1900,15 +1900,16 @@ public Collection<LoanScheduleAccrualData> retrivePeriodicAccrualData(final Loca
                 " where  (recaldet.is_compounding_to_be_posted_as_transaction is null or recaldet.is_compounding_to_be_posted_as_transaction = false) ")
                 .append(" and (((ls.fee_charges_amount <> COALESCE(ls.accrual_fee_charges_derived, 0))")
                 .append(" or (ls.penalty_charges_amount <> COALESCE(ls.accrual_penalty_charges_derived, 0))")
-                .append(" or (ls.interest_amount <> COALESCE(ls.accrual_interest_derived, 0)))")
+                .append(" or (ls.interest_amount > COALESCE(ls.interest_completed_derived, 0)")
+                .append(" and (ls.interest_amount <> COALESCE(ls.accrual_interest_derived, 0))))")
                 .append(" and loan.loan_status_id=:active and mpl.accounting_type=:type and (loan.closedon_date <= :tillDate or loan.closedon_date is null)")
                 .append(" and loan.is_npa=false and (ls.duedate <= :tillDate or (ls.duedate > :tillDate and ls.fromdate < :tillDate))) ");
         Map<String, Object> paramMap = new HashMap<>(4);
         if (organisationStartDate != null) {
             sqlBuilder.append(" and ls.duedate > :organisationStartDate ");
             paramMap.put("organisationStartDate", organisationStartDate);
         }
-        sqlBuilder.append(" order by loan.id,ls.duedate ");
+        sqlBuilder.append("  AND loan.id=15868 order by loan.id,ls.duedate ");
```

Comment:
Please remove the SQL statement for **_AND loan.id =15868_**

Reply:
@leonardohildt You are right it was by mistake Thanks

Reply:
Fixed now